### ISSUE TITLE
[Bug] No device tracking

### ISSUE THREAD

**Issue by skulumani** [tomaae/homeassistant-mikrotik_router#228]
## Describe the issue
Have Mikrotik router integration set up but no devices are being tracked.

## How to reproduce the issue
1. Install mikrotik router integration
2.  Use user with proper mikrotik permission
3. Enable "Track network devices" in configuration options
4. No device_tracker sensors appear within integration

## Expected behavior
Expect that all network devices show up within homeassistant

## Screenshots
![image](https://user-images.githubusercontent.com/5266022/183287690-be637e16-e380-43e2-b784-0b1b0914d593.png)

![image](https://user-images.githubusercontent.com/5266022/183287701-52c6cb45-f027-45d0-b0b3-3e4dbcc7304d.png)

![image](https://user-images.githubusercontent.com/5266022/183287707-35f38660-6e55-438d-860b-0190f79bbcd4.png)

## Software versions

 - Home Assistant version: 2022.7.7
 - Mikrotik Router integration version: 2.0
 - Mikrotik Hardware: hap AC2
 - RouterOS version:  6.49.6

## Diagnostics data

[config_entry-mikrotik_router-3ed14618eae654730434af485d991b52.json.txt](https://github.com/tomaae/homeassistant-mikrotik_router/files/9276648/config_entry-mikrotik_router-3ed14618eae654730434af485d991b52.json.txt)

## Traceback/Error logs
No logs

## Additional context

Seems to have only recently stopped tracking network devices. Perhaps a routeros update or Homeassistant?

**Comment by tomaae** [comment]
Are you sure there is nothing in the HA log? I can see device tracking information to be available in diagnostic data.
If you cannot find anything, I will need debugs to see if we can figure it out.

**Comment by skulumani** [comment]
[home-assistant.log](https://github.com/tomaae/homeassistant-mikrotik_router/files/9276965/home-assistant.log)

Here's a log after turning on debug logging for the integration. See this error for my two APs but none of the other devices.

~~~
2022-08-07 10:37:57 ERROR (MainThread) [homeassistant.util.logging] Exception in update_controller when dispatching 'mikrotik_router-update-Mikrotik': ()
Traceback (most recent call last):
  File "/config/custom_components/mikrotik_router/device_tracker.py", line 49, in update_controller
    update_items(
  File "/config/custom_components/mikrotik_router/device_tracker.py", line 90, in update_items
    trackers[item_id].async_schedule_update_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 677, in async_schedule_update_ha_state
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 525, in async_write_ha_state
    raise RuntimeError(f"Attribute hass is None for {self}")
RuntimeError: Attribute hass is None for <Entity Downstairs WAP: home>
~~~

**Comment by tomaae** [comment]
Looks like a conflict. First, try to remove the router from HA, make sure all entities are deleted, restart HA and readd the router.

**Comment by skulumani** [comment]
Thanks for the help. Removing mikrotik integration then restarting seemed to help.

Appreciate it! And the great integration

**Comment by skulumani** [comment]
Looks like the same issue is occurring. I've tried completely deleting the database volume in docker and recreating it, removing the integration and restarting HA, then re-add. 

Still seeing this error in the log

~~~
Logger: homeassistant.util.logging
Source: util/logging.py:167
First occurred: 12:12:44 AM (5 occurrences)
Last logged: 12:12:49 AM

Exception in update_controller when dispatching 'mikrotik_router-update-Mikrotik': () Traceback (most recent call last): File "/config/custom_components/mikrotik_router/model.py", line 42, in update_controller model_update_items( File "/config/custom_components/mikrotik_router/model.py", line 159, in model_update_items if tmp := _register_entity(sensors, item_id, uid, uid_sensor): File "/config/custom_components/mikrotik_router/model.py", line 76, in _register_entity _sensors[_item_id].async_schedule_update_ha_state() File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 677, in async_schedule_update_ha_state self.async_write_ha_state() File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 525, in async_write_ha_state raise RuntimeError(f"Attribute hass is None for {self}") RuntimeError: Attribute hass is None for <Entity Downstairs WAP: home>
~~~

Not sure if you have any other suggestions

**Comment by tomaae** [comment]
are you using 2.1? if so, run integration in debug and we will see what we can find out.

**Comment by skulumani** [comment]
Appreciate the help. I am on 2.1 now. Here's a log

[home-assistant.log](https://github.com/tomaae/homeassistant-mikrotik_router/files/9347980/home-assistant.log)

Also now it looks like HA won't boot

**Comment by tomaae** [comment]
I need you to run integration in debug, standard errors wont help here.

**Comment by skulumani** [comment]
Thanks I can do that. Is there documentation on how to enable debug mode for the integration?

I see on the main readme, instructions for debug logging but am unsure how to accomplish anything further.

Thanks

**Comment by tomaae** [comment]
thats all you need to do. debug logs will appear in ha log

**Comment by skulumani** [comment]
Here are some logs. Split it up into 450MB files

[log 0](https://send.bitwarden.com/#w0MhJ0Dj-kOOPK7zANwCFw/goo3V4OxyrroTdg2xS36fw)

**Comment by tomaae** [comment]
you didnt needed to run it that long :)
Can you give me entity info for "Galaxy S20"? Best would be directly from entity db, but if oyu dont know how, look for it in developer tools>states

**Comment by skulumani** [comment]
![image](https://user-images.githubusercontent.com/5266022/185280387-e2433437-c1f8-491e-be80-930403de44b5.png)

Does this work?

> you didnt needed to run it that long :)

I forgot about it overnight. The log was 2.5GB and I had a bunch of trouble even getting it open :)

**Comment by tomaae** [comment]
first 2 minutes are usually fine :)

is there no other sensor with similar name? that one does not belong to this integration.
If not, have a look for "Downstairs WAP". Thats another one raising an error.

**Comment by skulumani** [comment]
I'm not sure what is happening. I just renabled the "track network devices" option and it seems to be working now

![Screenshot_20220818-210528](https://user-images.githubusercontent.com/5266022/185520517-b9da206d-9476-4999-9054-bc26be459180.png)

**Comment by tomaae** [comment]
I have no idea what is happening here. Data seems to be perfectly fine.
Did you do any change in HA in between?
While this should be no issue, just to be sure, are you using any other integration to track devices on network?
Did any device left or joined the network? Could be wireless device.

That error coming from Galaxy S20 entity is weird one tho. Could you try it with that device active and using your wifi network (not in sleep mode) and another time with it turned off completly so it wont be connected for sure?

**Comment by skulumani** [comment]
I don't think I've made any changes. I have added/removed the integration
many times, and tried to clear/purge and deleted the entire database on
several occasions. The integration and HA was in the same state as the
logs, except I disabled and renabled the device tracking through the UI.

But the Galaxy S20 has been a constant issue for me. I think it uses some
kind of wifi sleep mode so it constantly disconnects and ruins my presence
detection.

I'll try to do some more testing with that device now that the device
tracking seems to be working. The only other device trackers I use are the
mobile apps, but those only provide data when on the local network so
they're much less useful than the network detection.

Now that it seems to be working I don't want to do anything in case I upset
the HA gods.
